doc/source/operations/rocky-linux-9.rst

```diff
@@ -0,0 +1,604 @@
+
+-  https://review.opendev.org/c/openstack/kayobe/+/898563 (to fix ``kayobe overcloud deprovision``)
+-  https://review.opendev.org/c/openstack/kayobe/+/898284 (if deployment predates Ussuri)
+
+   - TODO: Put this into the procedure.
+   -  **Must reprocess inspection data to update IPA kernel URL (see
+      release note)**
+
+-  https://review.opendev.org/c/openstack/kayobe/+/898777
+-  https://review.opendev.org/c/openstack/kayobe/+/898915
+-  https://review.opendev.org/c/openstack/kayobe/+/898905
+-  https://review.opendev.org/c/openstack/kolla/+/878835
+-  https://review.opendev.org/c/openstack/kayobe/+/898434 (if seeing slow fact gathering)
+-  https://review.opendev.org/c/openstack/kolla-ansible/+/900034
+-  https://review.opendev.org/c/openstack/kolla-ansible/+/897667
+-  https://review.opendev.org/c/openstack/nova/+/898554
+
+Configuration
+=============
+
+Make the following changes to your Kayobe configuration:
+
+- Merge in the latest ``stackhpc-kayobe-config`` ``stackhpc/yoga`` branch.
+- Set ``os_distribution`` to ``rocky`` in ``etc/kayobe/globals.yml``.
+- Set ``os_release`` to ``"9"`` in ``etc/kayobe/globals.yml``.
+- If you are using Kayobe multiple environments, add the following into
+  ``kayobe-config/etc/kayobe/environments/<env>/kolla/config/nova.conf``
+  (as Kolla custom service config environment merging is not supported in
+  Yoga). See `this PR
+  <https://github.com/stackhpc/stackhpc-kayobe-config/pull/648>`__ for details.
+
+  .. code-block:: ini
+
+     [libvirt]
+     hw_machine_type = x86_64=q35
+     num_pcie_ports = 16
+
+  This change does not need to be applied before migrating to Rocky Linux 9, but it should cause no harm to do so.
+  Note that this will not affect existing VMs, only newly created VMs.
+
+Prerequisites
+=============
+
+Before starting the upgrade, ensure any appropriate prerequisites are
+satisfied. These will be specific to each deployment, but here are some
+suggestions:
+
+* Ensure that there is sufficient hypervisor capacity to drain
+  at least one node.
+* If using Ironic for bare metal compute, ensure that at least one node is
+  available for testing provisioning.
+* Ensure that expected test suites are passing, e.g. Tempest.
+* Resolve any Prometheus alerts.
+* Check for unexpected ``ERROR`` or ``CRITICAL`` messages in Kibana/OpenSearch
+  Dashboard.
+* Check Grafana dashboards.
+
+Migrate to OpenSearch
+---------------------
+
+Elasticsearch/Kibana should be migrated to OpenSearch.
+
+- Read the `Kolla Ansible OpenSearch migration
+  docs <https://docs.openstack.org/kolla-ansible/yoga/reference/logging-and-monitoring/central-logging-guide-opensearch.html#migration>`__
+- If necessary, take a backup of the Elasticsearch data.
+- Ensure ``kolla_enable_elasticsearch`` is unset in ``etc/kayobe/kolla.yml``
+- Set ``kolla_enable_opensearch: true`` in ``etc/kayobe/kolla.yml``
+- ``kayobe kolla ansible run opensearch-migration``
```

Comment:
As `kayobe kolla ansible run` doesn't generate the local Kolla config, you need to run `kayobe overcloud service configuration generate --node-config-dir '/tmp/ignore' --kolla-tags none` before running the opensearch-migration.

Reply:
Would be good to recommend a `kayobe container image pull -kt opensearch` here, between the config gen and the migration. Just speeds up bringing up the new OpenSearch containers, so there's less of a downtime period.

Reply:
done